cxx-qt-lib/src/qcolor.rs

```diff
@@ -8,41 +8,41 @@
 #![allow(improper_ctypes)]
 
 use crate::actually_private::Private;
-use cxx::{type_id, ExternType};
+use cxx::{memory::UniquePtrTarget, type_id, ExternType};
 use std::{
+    ffi::c_void,
     marker::{PhantomData, PhantomPinned},
     mem::MaybeUninit,
     pin::Pin,
 };
 
-#[repr(u8)]
-#[allow(dead_code)]
-enum QColorSpec {
-    Unsupported = 0,
-    Rgb = 1,
-}
-
 extern "C" {
-    #[link_name = "cxxqt1$qcolor$init$from$argb"]
-    fn qcolor_init_from_argb(
-        this: &mut MaybeUninit<QColor>,
-        alpha: u16,
-        red: u16,
-        green: u16,
-        blue: u16,
+    #[link_name = "cxxqt1$qcolor$init$from$qcolor"]
+    fn qcolor_init_from_qcolor(ptr: &mut MaybeUninit<cxx::UniquePtr<QColor>>, qcolor: &QColor);
+    #[link_name = "cxxqt1$qcolor$init$from$rgba"]
+    fn qcolor_init_from_rgba(
+        ptr: &mut MaybeUninit<cxx::UniquePtr<QColor>>,
+        red: i32,
+        green: i32,
+        blue: i32,
+        alpha: i32,
     );
     #[link_name = "cxxqt1$qcolor$get$alpha"]
-    fn qcolor_get_alpha(this: &QColor) -> u16;
+    fn qcolor_get_alpha(this: &QColor) -> i32;
     #[link_name = "cxxqt1$qcolor$get$red"]
-    fn qcolor_get_red(this: &QColor) -> u16;
+    fn qcolor_get_red(this: &QColor) -> i32;
```

Comment:
funny how in C++ we call this parameter `self`, because `this` is a reserved keyword, and in Rust we use `this`, as `self` is reserved :smile: .
Anyway, good name.

Reply:
Hah !